### PR TITLE
check if is object before checking if method_exists

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
Pimcore X throws me that error on PHP8 when creating a brick programmatic:

```
Behat\Testwork\Call\Exception\FatalThrowableError: Type error: method_exists(): Argument #1 ($object_or_class) must be of type object|string, array given in vendor/pimcore/pimcore/models/DataObject/ClassDefinition.php:318
      Stack trace:
      #0 vendor/pimcore/pimcore/models/DataObject/ClassDefinition.php(318): method_exists(Array, 'getChildren')
      #1 vendor/pimcore/pimcore/models/DataObject/Objectbrick/Definition.php(210): Pimcore\Model\DataObject\ClassDefinition::cleanupForExport(Array)
      #2 vendor/pimcore/pimcore/models/DataObject/Objectbrick/Definition.php(180): Pimcore\Model\DataObject\Objectbrick\Definition->generateClassFiles(true)
      #3 src/CoreShop/Behat/Context/Setup/PimcoreClassContext.php(178): Pimcore\Model\DataObject\Objectbrick\Definition->save()
      #4 vendor/behat/behat/src/Behat/Testwork/Call/Handler/RuntimeCallHandler.php(110): CoreShop\Behat\Context\Setup\PimcoreClassContext->createBrickNamed('Attribute')
      #5 vendor/behat/behat/src/Behat/Testwork/Call/Handler/RuntimeCallHandler.php(64): Behat\Testwork\Call\Handler\RuntimeCallHandler->executeCall(Object(Behat\Behat\Definition\Call\DefinitionCall))
      #6 vendor/behat/behat/src/Behat/Testwork/Call/CallCenter.php(138): Behat\Testwork\Call\Handler\RuntimeCallHandler->handleCall(Object(Behat\Behat\Definition\Call\DefinitionCall))
      #7 vendor/behat/behat/src/Behat/Testwork/Call/CallCenter.php(96): Behat\Testwork\Call\CallCenter->handleCall(Object(Behat\Behat\Definition\Call\DefinitionCall))
      #8 vendor/behat/behat/src/Behat/Behat/Tester/Runtime/RuntimeStepTester.php(125): Behat\Testwork\Call\CallCenter->makeCall(Object(Behat\Behat\Definition\Call\DefinitionCall))
      #9 vendor/behat/behat/src/Behat/Behat/Tester/Runtime/RuntimeStepTester.php(73): Behat\Behat\Tester\Runtime\RuntimeStepTester->testDefinition(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\InitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\StepNode), Object(Behat\Behat\Definition\SearchResult), false)
      #10 vendor/behat/behat/src/Behat/Behat/Hook/Tester/HookableStepTester.php(74): Behat\Behat\Tester\Runtime\RuntimeStepTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\InitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\StepNode), false)
      #11 vendor/behat/behat/src/Behat/Behat/EventDispatcher/Tester/EventDispatchingStepTester.php(76): Behat\Behat\Hook\Tester\HookableStepTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\InitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\StepNode), false)
      #12 vendor/behat/behat/src/Behat/Behat/Tester/StepContainerTester.php(59): Behat\Behat\EventDispatcher\Tester\EventDispatchingStepTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\InitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\StepNode), false)
      #13 vendor/behat/behat/src/Behat/Behat/Tester/Runtime/RuntimeScenarioTester.php(76): Behat\Behat\Tester\StepContainerTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\InitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\ScenarioNode), false)
      #14 vendor/behat/behat/src/Behat/Behat/Hook/Tester/HookableScenarioTester.php(74): Behat\Behat\Tester\Runtime\RuntimeScenarioTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\InitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\ScenarioNode), false)
      #15 vendor/behat/behat/src/Behat/Behat/EventDispatcher/Tester/EventDispatchingScenarioTester.php(106): Behat\Behat\Hook\Tester\HookableScenarioTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\InitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\ScenarioNode), false)
      #16 vendor/behat/behat/src/Behat/Behat/Tester/Runtime/IsolatingScenarioTester.php(69): Behat\Behat\EventDispatcher\Tester\EventDispatchingScenarioTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\InitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\ScenarioNode), false)
      #17 vendor/behat/behat/src/Behat/Behat/Tester/Runtime/RuntimeFeatureTester.php(84): Behat\Behat\Tester\Runtime\IsolatingScenarioTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\UninitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\ScenarioNode), false)
      #18 vendor/behat/behat/src/Behat/Behat/Hook/Tester/HookableFeatureTester.php(72): Behat\Behat\Tester\Runtime\RuntimeFeatureTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\UninitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), false)
      #19 vendor/behat/behat/src/Behat/Behat/EventDispatcher/Tester/EventDispatchingFeatureTester.php(74): Behat\Behat\Hook\Tester\HookableFeatureTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\UninitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), false)
      #20 vendor/behat/behat/src/Behat/Testwork/Tester/Runtime/RuntimeSuiteTester.php(63): Behat\Behat\EventDispatcher\Tester\EventDispatchingFeatureTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\UninitializedSymfonyExtensionEnvironment), Object(Behat\Gherkin\Node\FeatureNode), false)
      #21 vendor/behat/behat/src/Behat/Testwork/Hook/Tester/HookableSuiteTester.php(73): Behat\Testwork\Tester\Runtime\RuntimeSuiteTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\UninitializedSymfonyExtensionEnvironment), Object(Behat\Testwork\Specification\GroupedSpecificationIterator), false)
      #22 vendor/behat/behat/src/Behat/Testwork/EventDispatcher/Tester/EventDispatchingSuiteTester.php(75): Behat\Testwork\Hook\Tester\HookableSuiteTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\UninitializedSymfonyExtensionEnvironment), Object(Behat\Testwork\Specification\GroupedSpecificationIterator), false)
      #23 vendor/behat/behat/src/Behat/Testwork/Tester/Runtime/RuntimeExercise.php(71): Behat\Testwork\EventDispatcher\Tester\EventDispatchingSuiteTester->test(Object(FriendsOfBehat\SymfonyExtension\Context\Environment\UninitializedSymfonyExtensionEnvironment), Object(Behat\Testwork\Specification\GroupedSpecificationIterator), false)
      #24 vendor/behat/behat/src/Behat/Testwork/EventDispatcher/Tester/EventDispatchingExercise.php(73): Behat\Testwork\Tester\Runtime\RuntimeExercise->test(Array, false)
      #25 vendor/behat/behat/src/Behat/Testwork/Ordering/OrderedExercise.php(80): Behat\Testwork\EventDispatcher\Tester\EventDispatchingExercise->test(Array, false)
      #26 vendor/behat/behat/src/Behat/Testwork/Tester/Cli/ExerciseController.php(149): Behat\Testwork\Ordering\OrderedExercise->test(Array, false)
      #27 vendor/behat/behat/src/Behat/Testwork/Tester/Cli/ExerciseController.php(108): Behat\Testwork\Tester\Cli\ExerciseController->testSpecifications(Object(Symfony\Component\Console\Input\ArgvInput), Array)
      #28 vendor/behat/behat/src/Behat/Testwork/Cli/Command.php(63): Behat\Testwork\Tester\Cli\ExerciseController->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #29 vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php(256): Behat\Testwork\Cli\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #30 vendor/symfony/symfony/src/Symfony/Component/Console/Application.php(971): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #31 vendor/symfony/symfony/src/Symfony/Component/Console/Application.php(290): Symfony\Component\Console\Application->doRunCommand(Object(Behat\Testwork\Cli\Command), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #32 vendor/behat/behat/src/Behat/Testwork/Cli/Application.php(124): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #33 vendor/symfony/symfony/src/Symfony/Component/Console/Application.php(166): Behat\Testwork\Cli\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #34 vendor/behat/behat/bin/behat(34): Symfony\Component\Console\Application->run()
      #35 {main}
```

### CODE REVIEW

File: models/DataObject/ClassDefinition.php [hunk 1]
```diff
@@ -315,7 +315,7 @@ public static function cleanupForExport(&$data)
             }
         }
 
-        if (method_exists($data, 'getChildren')) {
+        if (is_object($data) && method_exists($data, 'getChildren')) {
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition.php:318]: how about early return at line 301?
```php
        if (!is_object($data)) {
            return;
        }
```

Comment (dpfaffenbauer) [reply models/DataObject/ClassDefinition.php:318]: might as well work. should I change the PR?

Comment (dvesh3) [reply models/DataObject/ClassDefinition.php:318]: yes, please!

**Comment by dpfaffenbauer** [comment]
@dvesh3 done

**Comment by dvesh3** [comment]
@dpfaffenbauer thanks!

**Pull request merged**